core/src/main/scala/com/github/mjakubowski84/parquet4s/ParquetWriter.scala

```diff
@@ -29,21 +31,41 @@ trait ParquetWriter[T] {
     */
   def write(path: String, data: Iterable[T], options: ParquetWriter.Options)
 
+  /**
+    * Writes data out to given OutputFile
+    * @param outputFile a stream representation
+    * @param data data to write
+    * @param options configuration of how Parquet files should be created and written
+    */
+  def writeS(outputFile: OutputFile, data: Iterable[T], options: ParquetWriter.Options): Unit
+
 }
 
 
 object ParquetWriter  {
 
   private[parquet4s] type InternalWriter = HadoopParquetWriter[RowParquetRecord]
 
-  private class Builder(path: Path, schema: MessageType) extends HadoopParquetWriter.Builder[RowParquetRecord, Builder](path) {
+  private class BuilderPath(path: Path, schema: MessageType) extends HadoopParquetWriter.Builder[RowParquetRecord, BuilderPath](path) {
     private val logger = LoggerFactory.getLogger(ParquetWriter.this.getClass)
 
     if (logger.isDebugEnabled) {
       logger.debug(s"""Resolved following schema to write Parquet to "$path":\n$schema""")
     }
 
-    override def self(): Builder = this
+    override def self(): BuilderPath = this
+
+    override def getWriteSupport(conf: Configuration): WriteSupport[RowParquetRecord] = new ParquetWriteSupport(schema, Map.empty)
+  }
+
+  private class BuilderOutputFile(outputFile: OutputFile, schema: MessageType) extends HadoopParquetWriter.Builder[RowParquetRecord, BuilderOutputFile](outputFile) {
+    private val logger = LoggerFactory.getLogger(ParquetWriter.this.getClass)
+
+    if (logger.isDebugEnabled) {
+      logger.debug(s"""Resolved following schema to write Parquet to STream":\n$schema""")
```

Comment:
Remove the if check - this is macro-ed away at runtime

Reply:
This is base slf4j-api `Logger` so unfortunately, it isn't